src/libraries/Native/AnyOS/System.Globalization.Native/pal_icushim_internal.h

```diff
@@ -33,6 +35,21 @@
 #include <unicode/urename.h>
 #include <unicode/ustring.h>
 
+#else
+
+#include "icu.h"
+
+#ifndef __typeof
+#define __typeof decltype
```

Comment:
Hm, isn't `decltype` a C++ thing? (and not C)

Reply:
Yeah that's true. However in windows where this library is built builds using VC++. I didn't find a better way to do this given that we use this to declare the function pointers:
PER_FUNCTION_BLOCK(fn, lib) EXTERN_C __typeof(fn)* fn##_ptr;

is there a better way to do it on C like code for Windows?

Reply:
Sadly, I'm not aware of a better way... 
Ugh, it seems like there really isn't any standard way to get the type of something in C. (Seems there are many Stackoverflow posts for this too... [ex](https://stackoverflow.com/questions/14877415/difference-between-typeof-typeof-and-typeof-in-objective-c), [ex](https://stackoverflow.com/questions/29379172/best-way-to-emulate-typeof-for-msvc-or-alternative-workaround), etc..)

Since `__typeof` is a non-standard compiler extension it looks like `decltype` is the only way for MSVC.

Feel free to resolve/won't fix this comment on your PR. :)
(Though FWIW, you might add a brief comment that __typeof is non-standard, for the next person that looks at the code.)

Reply:
Sounds good. Thanks for the suggestion though. Yeah I saw the various posts but got to the same conclusion.